### ISSUE TITLE
Remove use of dc_array_get_raw

### ISSUE THREAD

**Issue by link2xt** [deltachat/deltachat-android#1013]
There is a related discussion in https://github.com/deltachat/deltachat-core-rust/pull/211

`dc_array_get_raw` is too low-level and relies on the internal representation of arrays. It can easily be replaced with repeated use of `dc_array_get_uint` so `dc_array_get_raw` can be deprecated.

**Comment by r10s** [comment]
dc_array_get_raw() was there for speed reasons mainly, esp. if the sizeof(array-element) is equal to sizeof(java-int) we could use a handy shortcut function, SetIntArrayRegion().

however, i agree that this is somewhat low-level and can be replaces. calling some thousands time through dc_array_get_uint() is probably fast enough. but i would continue preparing the array in memory and call into java only once using SetIntArrayRegion() - but this is unrelated to rust.

**Comment by link2xt** [comment]
Closing in favor of https://github.com/deltachat/deltachat-core-rust/pull/369

This will require only minor changes to the code

**Comment by r10s** [comment]
great that you push the forward. i commented in https://github.com/deltachat/deltachat-core-rust/pull/369
